packages/domain-search/src/components/search-bar/index.tsx

```diff
@@ -1,15 +1,27 @@
 import { __experimentalHStack as HStack } from '@wordpress/components';
 import { useI18n } from '@wordpress/react-i18n';
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import { useDomainSearch } from '../../page/context';
 import { DomainSearchControls } from '../../ui';
+import { Filter } from './filter';
+import { FilterState } from './types';
 
 const DELAY_TIMEOUT = 300;
 
+const emptyFilter: FilterState = {
+	exactSldMatchesOnly: false,
+	tlds: [],
+};
+
 export const SearchBar = () => {
 	const { __ } = useI18n();
 	const { query, setQuery } = useDomainSearch();
 	const [ localQuery, setLocalQuery ] = useState( query );
+	const [ filter, setFilter ] = useState( emptyFilter );
```

Comment:
We need to move these two to the context as they're going to be used in the suggestion query.